src/main.rs

```diff
@@ -22,36 +24,77 @@ async fn main() -> Result<(), AppError> {
         env!("GIT_REVISION")
     );
 
-    let (app, _, icebreakers_api, api_prefix) = build(config.clone().into()).await?;
+    let (app, _, health_monitor, icebreakers_api, api_prefix) =
+        build(config.clone().into()).await?;
+
     let address = std::net::SocketAddr::new(config.server_address, config.server_port);
     let listener = tokio::net::TcpListener::bind(address).await?;
-    let (ready_tx, ready_rx) = oneshot::channel();
     let shutdown_signal = async {
-        let _ = signal::ctrl_c().await;
+        let _ = tokio::signal::ctrl_c().await;
         info!("Received shutdown signal");
     };
 
+    let notify_server_ready = Arc::new(tokio::sync::Notify::new());
+
     // Spawn the server in its own task
     let spawn_task = tokio::spawn({
-        let app = app;
+        let notify_server_ready = notify_server_ready.clone();
+        let app = app.clone();
         async move {
-            let server_future =
-                axum::serve(listener, ServiceExt::<Request>::into_make_service(app))
-                    .with_graceful_shutdown(shutdown_signal);
+            let server_future = axum::serve(listener, app.into_make_service())
+                .with_graceful_shutdown(shutdown_signal);
 
             // Notify that the server has reached the listening stage
-            let _ = ready_tx.send(());
+            notify_server_ready.notify_one();
 
             server_future.await
         }
     });
 
-    if let Ok(()) = ready_rx.await {
-        info!("Server is listening on http://{}{}", address, api_prefix);
```

Comment:
Optional: We might want to also mention a log about ws

Reply:
Similar response:

The URI(s) of load balancer(s) will be logged soon after this:

https://github.com/blockfrost/blockfrost-platform/blob/2f4cab53186e39c3db6d775777d80a14658011ac/src/load_balancer.rs#L273